technical_docs/Verification.md

```diff
@@ -0,0 +1,94 @@
+2. If the current effective UID has a valid home directory, `~/.scitokens/vo.example.com`
+3. `/etc/scitokens/vo.example.com`
+
+To establish a set of trust roots, first establish a list of search files using the following pseudo-code algorithm:
+
+```
+search_files = []
+for dir_path in search_path:
+   files = list_valid(dir_path)
+   sort(files)
+   for file in files:
+       search_files.append(file)
+```
+
+* `list_valid` lists all the files in a given directory NOT matching the regex ` ^((\..*)|(.*~)|(#.*)|(.*\.rpmsave)|(.*\.rpmnew)|(.*\.dpkg-old)|(.*\.dpkg-dist)|(.*\.cfsaved))$`
+* The `sort` function sorts the file names in lexigraphical order.
+
+For each file in the search path:
+
+* If the filename ends in `.jwks`, its contents should be parsed as a JSON object representing a JSON Web Key Set as specified by section 5 of RFC 7517.
+* If the filename ends in `.jku`, then the file contents should be parsed as a JWK Set URL as specified in section 4.1.2 of RFC 7515.  This mechanism allows the VO (or a delegate organization) to post a set of keys that should be used as trust roots (assisting with key rotation).
+   * If a `.jku` file is used, an implementation MAY cache the set of keys locally.  If the keys are retrieved via a HTTP GET request and the server responds with a `Cache-Control` header as defined by RFC 7234, then the implementation MUST use this expire time.  Otherwise, max object age in the cache is implementation-defined.  An implementation SHOULD default the max age to less than 24 hours.
+* Otherwise, the file is ignored.  Additional parsing rules may be added at a later date.
+
+The above algorithm results in an ordered list of JSON Web Keys corresponding to each known VO name; these will be used in the next section.
+
+Note the VO name is used as part of a path name in the validation; it is recommended the VO name be based on a known DNS name (such as `vo.example.com`), although historical names may be used for existing communities (`cms`, `atlas`, `ligo`, etc).  Case-sensitive string-matching rules from JSON should be used when doing the directory lookup.  (Author's note: this definition of VO name appears quite problematic.  We want to have a human-friendly name such as `ligo`, but really would benefit from having a global unambiguous identifier such as a URL).
+
+Verifying a SciToken
+--------------------
+
+A SciToken MUST be a properly-formatted [JSON Web Token](https://jwt.io) (JWT), as described by [RFC 7519](https://tools.ietf.org/html/rfc7519).
+
+The SciToken MUST be signed with an asymmetric key; the public portion of the key MUST be determined by the value of least one of the following JOSE header attributes:
+* `pwt`: The parent web token (TODO: write up separate document and link to this), a SciToken used to sign another SciToken.  The parent web token itself MUST be verified according to this document.  The `pwt` MUST specify a "child web key" (`cwk`) and the SciToken MUST be signed by that `cwk`.
```

Comment:
I'm concerned we're quickly going off the standards track. Until we have a clear need for token delegation, I want to focus on using RFC-compliant [JWS signature validation](https://tools.ietf.org/html/rfc7515#section-5.2)